v2/writer.go

```diff
@@ -0,0 +1,125 @@
+package car
+
+import (
+	"bytes"
+	"context"
+	"github.com/ipfs/go-cid"
+	format "github.com/ipfs/go-ipld-format"
+	car_v1 "github.com/ipld/go-car"
+	"github.com/willscott/carbs"
+	"io"
+)
+
+type (
+	// Padding represents the number of padding bytes.
+	Padding uint64
+	// Writer writes CAR v2 into a give io.Writer.
+	Writer struct {
+		Walk         car_v1.WalkFunc
+		IndexCodec   carbs.IndexCodec
+		NodeGetter   format.NodeGetter
+		CarV1Padding Padding
+		IndexPadding Padding
+
+		ctx   context.Context
+		roots []cid.Cid
+	}
+)
+
+// WriteTo writes this padding to the given writer as default value bytes.
+func (p Padding) WriteTo(w io.Writer) (n int64, err error) {
+	paddingBytes := make([]byte, p)
```

Comment:
I disagree with using `make([]byte, N)` for padding; if I instruct 100MiB of padding because I expect to write lots of blocks, should that really allocate that much at once?

If you're implementing a WriteTo, at least you can be more clever here, such as writing small chunks of zeros until you've written N bytes, via something like `var zero [1024]byte` and a loop.

Or, if you prefer something a bit higher level, you could look at combining `io.CopyN` with a reader that just keeps giving zeros, but I imagine that would end up being more lines anyway.